### ISSUE TITLE
presubmission : fgczgseaora

### ISSUE THREAD

**Issue by wolski** [ropensci/software-review#337]
Submitting Author: Witold Wolski (@wolski)  
Repository:   https://github.com/protViz/fgczgseaora

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: fgczgseaora
Type: Package
Title: ORA, sigORA and GSEA functionalities for proteomic data
Version: 0.1.0
Authors@R: c(person("Lucas", "Kook",
    email = "lucasheinrich.kook@uzh.ch",
    role = c("aut"),
    comment = c(ORCID = "0000-0002-7546-7356")),
    person("Witold E.", "Wolski",
    email = "wew@fgcz.ethz.ch",
    role = c("cre"),
    comment = c(ORCID = "0000-0002-6468-120X")))
Description: Implements gene pair signature based over representation
    analysis and gene set enrichment tools. Allows background specification
    for sigORA using the sigora package  without further ID mapping being necessary.
Depends: R (>= 3.5), WebGestaltR (>= 0.4)
License: GPL-2
Encoding: UTF-8
LazyData: true
Imports: AnnotationDbi,
    GO.db,
    S4Vectors,
    UpSetR,
    dplyr,
    DT,
    ggplot2,
    magrittr,
    org.Hs.eg.db,
    quantable,
    reactome.db,
    readr,
    rlang,
    sigora,
    tibble,
    tidyr,
    tidyverse
RoxygenNote: 6.1.1
VignetteBuilder: knitr
Suggests: knitr, rmarkdown, bookdown, slam, BiocStyle, docopt
biocViews: MassSpectrometryData, ReproducibleResearch
```

## Scope 

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below.:

	- [ ] data retrieval
	- [ ] data extraction
	- [x] database access
	- [ ] data munging
	- [ ] data deposition
	- [x] reproducibility
	- [ ] geospatial data
	- [ ] text analysis
	

- Explain how the and why the package falls under these categories (briefly, 1-2 sentences).  Please note any areas you are unsure of:

A plethora of R packages exist on CRAN and Bioconductor to perform over-representation 
analysis (ORA) and gene set enrichment analysis (GSEA). However, consistency in the 
underlying nomenclature for specific analyses and user friendly implementation is 
still lacking. `fgczgseaora` aims at unifying ID mapping and enrichment analysis 
in a syntactically coherent and intuitive way, while ensuring reproducibility of 
results. `fgczgseaora` primarily consists of wrapper functions around the 
`r CRANpkg("sigora")` and `r CRANpkg("WebGestaltR")` packages from CRAN and 
`r CRANpkg("rmarkdown")` based reports for visualisation and contextualisation 
 of analysis results.

-   Who is the target audience and what are scientific applications of this package?  

Biologist and Bioinformaticians working with proteomics data.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

See two bullet points above.

-  Any other questions or issues we should be aware of?:

**Comment by melvidoni** [comment]
Hello @wolski, thanks for your submission. Before we can continue, your opening issue is incomplete. We need it complete in order to be start the process. Please, edit your post and add the following at the bottom (adding your replies, of course). Let me know when it is done.

```
## Technical checks

Confirm each of the following by checking the box.  This package:

- [ ] does not violate the Terms of Service of any service it interacts with. 
- [ ] has a CRAN and OSI accepted license.
- [ ] contains a [README with instructions for installing the development version](https://ropensci.github.io/dev_guide/building.html#readme). 
- [ ] includes [documentation with examples for all functions](https://ropensci.github.io/dev_guide/building.html#documentation).
- [ ] contains a vignette with examples of its essential functions and uses.
- [ ] has a [test suite](https://ropensci.github.io/dev_guide/building.html#testing).
- [ ] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage using services such as Travis CI, Coveralls and/or CodeCov.

## Publication options

- [ ] Do you intend for this package to go on CRAN?  
- [ ] Do you wish to automatically submit to the [Journal of Open Source Software](http://joss.theoj.org/)? If so:

<details>
 <summary>JOSS Options</summary>  

  - [ ] The package has an **obvious research application** according to [JOSS's definition](https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements).
    - [ ] The package contains a `paper.md` matching [JOSS's requirements](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) with a high-level description in the package root or in `inst/`.
    - [ ] The package is deposited in a long-term repository with the DOI: 
    - (*Do not submit your package separately to JOSS*)  
  
</details>

- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:  

<details>
<summary>MEE Options</summary>  

- [ ] The package is novel and will be of interest to the broad readership of the journal. 
- [ ] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [ ] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.github.io/dev_guide/policies.html#code-of-conduct) during the review process and in maintaining my package should it be accepted.
```

**Comment by wolski** [comment]
Hello @melvidoni 

This is a "presubmission inquiry".
I would expect to find an answer to question:
Not sure your software fits? 

in your reply.

See also this link if you are not sure what I mean.
https://github.com/ropensci/software-review/issues/new/choose

Have a great day
Witek

**Comment by melvidoni** [comment]
Thank you @wolski for bringing your package to rOpenSci. If this is a presubmission, then kindly add that to the title of the issue, as it is expected. I will discuss this package with the other editors.

**Comment by melvidoni** [comment]
Hello @wolski. We have discussed your presubmission, but we found that your package is out of scope, as it doesn't fit completely on the categories we have for now. I do hope you can find another venue, as the package is interesting and useful. Thank you for submitting, and apologies for the original confusion due to the issue name.

**Comment by wolski** [comment]
Hello @melvidoni,

I did notice that since my submission a new category was introduced i.e. 
- [ ] scientific software wrappers

Which I believe our package fgcz.gsea.ora fit's rather well.
Could you please have a second look at our application?

**Comment by melvidoni** [comment]
Hello @wolski. @geanders can you please take a look at this?

**Comment by geanders** [comment]
Yes, I will look into this. Thanks @melvidoni!

**Comment by geanders** [comment]
Thank you for your continued interest, @wolski . I have re-evaluated, and this package would still fall out of scope, even with the introduction of the new category of "scientific software wrappers". This category captures packages that wrap software that is available outside of the R environment, but not other R packages. There are some more details on this in the [ROpenSci book's section on Aims and Scope](https://devguide.ropensci.org/policies.html#aims-and-scope): 

> "**scientific software wrappers:** Packages that wrap non-R utility programs used for scientific research. These programs must be specific to research fields, not general computing utilities. Wrappers must be non-trivial, in that there must be significant added value above simple system() call or bindings, whether in parsing inputs and outputs, data handling, etc. Improved installation process, or extension of compatibility to more platforms, may constitute added value if installation is complex. This does not include wrappers of other R packages or C/C++ libraries that can be included in R packages. We strongly encourage wrapping open-source and open-licensed utilities - exceptions will be evaluated case-by-case, considering whether open-source options exist."

It looks like the `fgczgseaora` package does wrap other software, but these are R packages, and so it does not fit into this new category, which focuses on providing access to functionality otherwise not available in R. I apologize that I do not have better news for you, but I wish you good luck in continuing work on developing and promoting your package.
